File: libs/summary/domain/entities/source-mix-entry.ts
export type SourceMixEntry = {
  readonly providerKey: string;
  readonly itemCount: number;
  readonly citationCount: number;
  readonly storyClusterCount: number;
  readonly crossSourceClusterCount: number;
  readonly singleSourceOnly: boolean;
  readonly topicIds: readonly string[];
};

File: apps/frontend/features/summaries/lib/src/domain/entities/source_mix_entry.dart
final class SourceMixEntry {
  const SourceMixEntry({
    required this.providerKey,
    required this.itemCount,
    required this.citationCount,
    required this.storyClusterCount,
    required this.crossSourceClusterCount,
    required this.singleSourceOnly,
    required this.interestIds,
  });

  final String providerKey;
  final int itemCount;
  final int citationCount;
  final int storyClusterCount;
  final int crossSourceClusterCount;
  final bool singleSourceOnly;
  final List<String> interestIds;
}
